Descent3/gamesave.cpp

```diff
@@ -399,8 +399,10 @@ void SaveGameDialog() {
   // create savegame directory if it didn't exist before.
   std::error_code ec;
   if (!std::filesystem::create_directories(savegame_dir, ec)) {
-    DoMessageBox(TXT_ERROR, TXT_ERRCREATEDIR, MSGBOX_OK);
-    return;
+    if (ec) {
+      DoMessageBox(TXT_ERROR, TXT_ERRCREATEDIR, MSGBOX_OK);
+      return;
+    }
   }
```

Comment:
We can check `ec` alone here, without checking return code of `create_directories`.